packages/core/test/pseudo-states.spec.ts

```diff
@@ -626,6 +626,58 @@ describe('pseudo-states', () => {
                 });
             });
 
+            it('should warn when pseudo-class expects params but none was given', () => {
+                const config = {
+                    entry: `/entry.st.css`,
+                    files: {
+                        '/entry.st.css': {
+                            namespace: 'entry',
+                            content: `
+                            .my-class {
+                                -st-states: state1(string);
+                            }
+                            |.my-class:state1 {}
+                            `,
+                        },
+                    },
+                }
+
+                const res = expectWarningsFromTransform(config, [
+                    {
+                        message: 'pseudo-state "state1" expects a parameter of type string but none was given',
+                        file: '/entry.st.css',
+                        severity: 'warning',
+                    }
+                ])
+
+                expect(res).to.have.styleRules({
+                    1: '.entry__my-class.entry---state1-0- {}',
+                });
+            })
+
+            it('should not warn when pseudo-class does not expect a params', () => {
+                const config = {
+                    entry: `/entry.st.css`,
+                    files: {
+                        '/entry.st.css': {
+                            namespace: 'entry',
+                            content: `
+                            .my-class {
+                                -st-states: state1;
+                            }
+                            .my-class:state1 {}
+                            `,
+                        },
+                    },
+                }
+
+                const res = expectWarningsFromTransform(config, [])
```

Comment:
I would expect from the title of the test that the test would try to call the state, that was defined with no param, with a param

Reply:
Also I would say that generally we try not to test things that should "not happen"

Reply:
I agree with that, but how would you make sure it won't happen? @tomrav @idoros

Reply:
In this case I probably wouldn't, and add a regression test only if it pops up somewhere unexpected somewhere down the line

Reply:
I believe there are plenty of non param tests that check for diagnostics that would break if this new one would be reported for them.

Also, slightly related, I think it would be good to have a test that non param states that are called with a param also report an error (if non exist already)

Reply:
I moved it to be a regression test.